src/dispatch/function.py

```diff
@@ -40,21 +50,17 @@ def __init__(
         client: Client,
         name: str,
         primitive_func: PrimitiveFunctionType,
-        func: Callable[P, T] | None,
-        coroutine: bool = False,
+        func: Callable[..., Any] | None,
```

Comment:
Do you think it would be worth splitting the `Function` type in two with a dedicated `PrimitiveFunction` to avoid this `None` and the check in `__call__`?

Reply:
See https://github.com/stealthrocket/dispatch-py/pull/101/commits/a9148a1352e9f0c8441d9bb130f27239ea99b5c8